plugin/core/src/main/groovy/com/novoda/gradle/release/MavenPublicationAttachments.groovy

```diff
@@ -0,0 +1,42 @@
+package com.novoda.gradle.release
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.component.SoftwareComponent
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.javadoc.Javadoc
+
+class MavenPublicationAttachments {
+
+    private final SoftwareComponent softwareComponent
+    private final List<Object> allArtifactSources
+
+    MavenPublicationAttachments(SoftwareComponent softwareComponent, def ... allArtifactSources) {
+        this(softwareComponent, Arrays.asList(allArtifactSources).asImmutable())
+    }
+
+    MavenPublicationAttachments(SoftwareComponent softwareComponent, List<Object> allArtifactSources) {
+        this.softwareComponent = softwareComponent
+        this.allArtifactSources = allArtifactSources
+    }
+
+    final void attachTo(MavenPublication publication) {
+        allArtifactSources.each { publication.artifact it }
+        publication.from softwareComponent
+    }
+
+    protected static Task sourcesJarTask(Project project, String publicationName, def ... sourcePaths) {
+        return project.task("genereateSourcesJarFor${publicationName.capitalize()}Publication", type: Jar) { Jar jar ->
```

Comment:
It's worth noting that I have also introduced what is a potential breaking change, as I have decided to enforce a more consistent naming for the tasks generating the sources and javadocs jars.